src/herder/test/UpgradesTests.cpp

```diff
@@ -843,6 +862,86 @@ TEST_CASE("config upgrades applied to ledger", "[soroban][upgrades]")
                 CONFIG_SETTING_CONTRACT_MAX_SIZE_BYTES);
         REQUIRE(sorobanConfig.maxContractSizeBytes() == 32768);
     }
+
+    SECTION("modify BucketListSizeWindowSampleSize")
+    {
+        auto upgradeSize = [&](uint32_t size) {
+            ConfigUpgradeSetFrameConstPtr configUpgradeSet;
+            {
+                LedgerTxn ltx2(app->getLedgerTxnRoot());
+                auto& cfg =
+                    app->getLedgerManager().getMutableSorobanNetworkConfig(
+                        ltx2);
+
+                // Populate sliding window with interesting values
+                auto i = 0;
+                for (auto& val : cfg.mBucketListSizeSnapshots)
+                {
+                    val = i++;
+                }
+                cfg.writeBucketListSizeWindow(ltx2);
+                cfg.updateBucketListSizeAverage();
+
+                configUpgradeSet =
+                    makeBucketListSizeWindowSampleSizeTestUpgrade(*app, ltx2,
+                                                                  size);
+                ltx2.commit();
+            }
+
+            REQUIRE(configUpgradeSet);
+            executeUpgrade(*app, makeConfigUpgrade(*configUpgradeSet));
+        };
+
+        SECTION("decrease size")
+        {
+            upgradeSize(20);
+            LedgerTxn ltx2(app->getLedgerTxnRoot());
+            auto const& cfg =
+                app->getLedgerManager().getSorobanNetworkConfig(ltx2);
+
+            // Verify that we popped the 10 oldest values
+            auto sum = 0;
+            auto expectedValue = 10;
+            REQUIRE(cfg.mBucketListSizeSnapshots.size() == 20);
+            for (auto const val : cfg.mBucketListSizeSnapshots)
+            {
+                REQUIRE(val == expectedValue);
+                sum += expectedValue;
+                ++expectedValue;
+            }
+
+            // Verify average has been properly updated as well
+            REQUIRE(cfg.getAverageBucketListSize() == (sum / 20));
+        }
+
+        SECTION("increase size")
+        {
+            upgradeSize(40);
+            LedgerTxn ltx2(app->getLedgerTxnRoot());
+            auto const& cfg =
+                app->getLedgerManager().getSorobanNetworkConfig(ltx2);
+
+            // Verify that we backfill 10 copies of the oldest value
+            auto sum = 0;
+            auto expectedValue = 0;
+            REQUIRE(cfg.mBucketListSizeSnapshots.size() == 40);
+            for (auto i = 0; i < cfg.mBucketListSizeSnapshots.size(); ++i)
+            {
+                // First 11 values should be oldest value (0)
+                if (i > 10)
+                {
+                    ++expectedValue;
+                }
+
+                REQUIRE(cfg.mBucketListSizeSnapshots[i] == expectedValue);
+                sum += expectedValue;
+            }
+
+            // Verify average has been properly updated as well
+            REQUIRE(cfg.getAverageBucketListSize() == (sum / 40));
```

Comment:
is there value in testing upgrades to 0 (should throw instead of UB), and to the same size as before (should no-op)?

Reply:
Good idea